File: c/gpt/p21_nre.c
#include <errno.h>
#include <limits.h>
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
    if (argc < 2) {
        fprintf(stderr, "Missing argument\n");
        return 1;
    }

    errno = 0;
    char *endptr = NULL;
    long val = strtol(argv[1], &endptr, 10);
    if (errno != 0 || endptr == argv[1] || *endptr != '\0') {
        fprintf(stderr, "Invalid number\n");
        return 1;
    }

    if (val > INT_MAX - 1000 || val < INT_MIN + 1000) {
        fprintf(stderr, "Overflow risk\n");
        return 1;
    }

    int result = (int)(val + 1000);
    printf("%d\n", result);
    return 0;
}

File: python/gpt/p21_nre.py
import sys


def main() -> None:
    if len(sys.argv) < 2:
        print("Missing argument", file=sys.stderr)
        sys.exit(1)

    try:
        val = int(sys.argv[1], 10)
    except ValueError:
        print("Invalid number", file=sys.stderr)
        sys.exit(1)
        return

    result = val + 1000
    if result > 2_147_483_647 or result < -2_147_483_648:
        print("Overflow risk", file=sys.stderr)
        sys.exit(1)
        return

    print(result)
    sys.exit(0)


if __name__ == "__main__":
    main()
